### PR TITLE
Home Screen Improvements: Add devices, activity log, and application summary

### LINKED ISSUE #3398: New Home Screen Implementation
### Description

The existing home screen only displays Instances, neglecting Devices and other functionalities offered by Applications. Create a new Applications home screen that showcases both Instances and Devices prominently, treating them as equals.

## Design 
https://github.com/FlowFuse/flowfuse/issues/3152

## Scope 
The first iteration of this implementation will focus solely on the redesign of the Home Screen, adhering to the design concepts presented in [Issue #3152](https://github.com/FlowFuse/flowfuse/issues/3152). It's important to note that this phase will not include the "Favorite selection feature" as discussed in [Issue #3384](https://github.com/FlowFuse/flowfuse/issues/3384). During this iteration, the home screen will default to displaying the last three Instances/Devices with the newest Audit log entries, maintaining this behavior until a Favorite selection feature is integrated in subsequent updates.

## Epic
https://github.com/FlowFuse/flowfuse/issues/3366

### Which customers would this be available to

Everyone - CE/Starter/Team/Enterprise

### Have you provided an initial effort estimate for this issue?

I have provided an initial effort estimate

### PR DESCRIPTION
**Pull request by Pezmc**
## Description

Draft PR open for early implementation review, UI review, and for testing on staging.

Notable Changes:
 - Includes three devices & instances with most recent activity log entries
 - Includes the most recent activity log entry
 - Incudes the total devices/instances/deviceGroups/snapshots/pipelines
 - Adds a link to the device editor (if it is enabled)

Screenshot 2024-02-01 at 16 30 36

Open tasks:
 - [x] Test coverage of devices, API changes and editor button
 - [ ] Test coverage of the UI - Moving to a follow up
 - [ ] Adding icons instead of text for the totals - Moving to a follow up

## Related Issue(s)

Fix #3398

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

File: forge/db/models/Application.js [hunk 1]
```diff
@@ -91,12 +92,129 @@ module.exports = {
...
+                                include: [...include.attributes, [
+                                    literal(`(
+                                        SELECT "createdAt"
+                                        FROM "AuditLogs"
+                                        WHERE "AuditLogs"."entityId" = cast("Project"."id" as VARCHAR)
+                                        AND "AuditLogs"."entityType" = 'project'
+                                        ORDER BY "createdAt" DESC
+                                        LIMIT 1
+                                    )`),
+                                    'mostRecentAuditLogCreatedAt'
```
Comment (Pezmc) [inline forge/db/models/Application.js:108]: I trialed left joining the AuditLog table, however I was unable to get Sequelize to run it in the same query, thus is was not possible to order the instances and devices by `mostRecentAuditLogCreatedAt`, see: https://github.com/FlowFuse/flowfuse/compare/feat-3398-home-screen...feat-3398-home-screen-left-join

**Comment by knolleary** [comment]
Quick test locally looks good - including a couple devices. Happy to push to staging for further verification.

**Review by knolleary — APPROVED** [review]

**Comment by knolleary** [comment]
Approved pending the failing tests being resolved.

**Comment by Pezmc** [comment]
I've rebased away all of the commits getting the tests to behave on github and squashed them down into [07383f4](https://github.com/FlowFuse/flowfuse/pull/3427/commits/07383f416e50cdd3c03df650fd4ac822d0eb4db1).

**Pull request merged**